airbyte-webapp/src/views/Connection/CatalogDiffModal/CatalogDiffModal.tsx

```diff
@@ -0,0 +1,42 @@
+import { useMemo } from "react";
+import { FormattedMessage } from "react-intl";
+
+import { Button } from "components";
+
+import { AirbyteCatalog, CatalogDiff } from "core/request/AirbyteClient";
+
+import { ModalBody, ModalFooter } from "../../../components/Modal";
+import styles from "./CatalogDiffModal.module.scss";
+import { DiffSection } from "./components/DiffSection";
+import { FieldSection } from "./components/FieldSection";
+import { getSortedDiff } from "./utils/utils";
+
+interface CatalogDiffModalProps {
+  catalogDiff: CatalogDiff;
+  catalog: AirbyteCatalog;
+  onClose: () => void;
+}
+
+export const CatalogDiffModal: React.FC<CatalogDiffModalProps> = ({ catalogDiff, catalog, onClose }) => {
```

Comment:
Could this whole modal be easily tested in a storybook with mock data?

Reply:
Yes, but it will require [mocking the context](https://storybook.js.org/docs/react/writing-stories/decorators#context-for-mocking) as this component is only ever rendered using the modal service.

I think it's worth adding and starting to add some of our services/contexts to storybook to expand our library, but it feels like a separate PR given the scope of the work.

Reply:
Spent some time with this Wednesday -- not easily, but I can continue working on it (maybe in a separate pr?)